### DISCUSSION TITLE
Help setting up Konnected Interface Panel with Homebridge and Hubitat

### DISCUSSION THREAD

**Discussion by lnjustin** [mkormendy/homebridge-konnected#27] [Q&A] [answered]
Is the homebridge konnected plugin useful at all if I'm using the Konnected Interface panel? 

I've tried setting it up, but the only thing that shows up in HomeKit is a "Konnected Alarm" device and it continually says "Disarming..." with a spinning wheel. No devices show up for any of the zones that I defined in the plugin's UI. So no doors for any of the contact sensors, no smoke detector, etc. Any ideas?

Things from other forum posts that I've tried, but that did not resolve the issue:
- updated each Konnected panel to the latest firmware
- made sure that I did not select anything under Triggerable Modes (Home, Away, etc.) for switches (with the yellow warning signs)
- Cleared the cache numerous times
- Rebooted Homebridge numerous times

Here's the status output from Homebridge upon restart:
[12/9/2022, 3:28:13 PM] Registering platform 'homebridge-konnected.konnected'
[12/9/2022, 3:28:13 PM] [Konnected] Loaded homebridge-konnected v1.3.1 child bridge successfully
[12/9/2022, 3:28:13 PM] Loaded 1 cached accessories from cachedAccessories.0EA35C301A29.
[12/9/2022, 3:28:14 PM] [Konnected] Loading accessory from cache: Konnected Alarm
[12/9/2022, 3:28:14 PM] [Konnected] Updating existing accessory: Konnected Alarm
[12/9/2022, 3:28:14 PM] Homebridge v1.6.0 (HAP v0.11.0) (Konnected) is running on port 44358.
[12/9/2022, 3:28:14 PM] [Konnected] Listening for zone changes on 192.168.1.91 port 56864

Config file in relevant part:
```
{
            "name": "Konnected",
            "advanced": {
                "exitDelaySettings": {
                    "audibleBeeperModes": [
                        "1"
                    ]
                }
            },
            "panels": [
                {
                    "name": "Konnected V1-V2",
                    "uuid": "XXXXXX",
                    "ipAddress": "192.168.1.44",
                    "port": 22261,
                    "blink": true,
                    "zones": [
                        {
                            "enabled": true,
                            "zoneNumber": "1",
                            "zoneType": "contact",
                            "zoneLocation": "Windows"
                        },
                        {
                            "enabled": false,
                            "zoneNumber": "out",
                            "zoneType": "strobe",
                            "zoneLocation": "Security System"
                        }
                    ]
                },
                {
                    "name": "Konnected V1-V2",
                    "uuid": "XXXXX",
                    "ipAddress": "192.168.1.42",
                    "port": 13881,
                    "blink": true,
                    "zones": [
                        {
                            "enabled": true,
                            "zoneNumber": "1",
                            "zoneType": "smoke",
                            "zoneLocation": "Smoke"
                        },
                        {
                            "enabled": true,
                            "zoneNumber": "2",
                            "zoneType": "contact",
                            "zoneLocation": "Front Door"
                        },
                        {
                            "enabled": true,
                            "zoneNumber": "3",
                            "zoneType": "contact",
                            "zoneLocation": "Mud Room Door"
                        },
                        {
                            "enabled": true,
                            "zoneNumber": "4",
                            "zoneType": "contact",
                            "zoneLocation": "Garage and Crawl Door"
                        },
                        {
                            "enabled": true,
                            "zoneNumber": "5",
                            "zoneType": "contact",
                            "zoneLocation": "Screened Porch Door"
                        },
                        {
                            "enabled": true,
                            "zoneNumber": "6",
                            "zoneType": "motion",
                            "zoneLocation": "Kitchen Motion"
                        }
                    ]
                }
            ],
            "_bridge": {
                "username": "XXXXX",
                "port": 44358
            },
            "platform": "konnected"
        }
```

EDIT: I realize that this post may mention two separate issues: (1) the spinning 'disarming' status; and (2) no devices showing up. I resolved (1) by your instructions in a previous post on this discussion board. But (2) remains unresolved.

Also, FYI I have Konnected integrated with Hubitat already for a long time now, but do not connect Konnected to HomeKit via Hubitat. My desired setup is to have two separate direct integrations, where Konnected is connected directly to Hubitat and is also connected to HomeKit via Homebridge. The primary motivator for this is that Konnected does not display a native alarm switch in HomeKit when integrated via Hubitat - if integration with both Hubitat and Homebridge simultaneously is a problem, any insight into how to get a native alarm switch in HomeKit would be helpful!

**Comment by mkormendy** [comment]
It is impossible at this time with the Konnected device's firmware to be controlled by two separate platforms. It's either one or the other.

**Reply by lnjustin** [reply]
Ah, bummer. I guess I'll keep pursuing trying to figure out how to get Konnected to show up in HomeKit right via Hubitat then. Thanks.

**Comment by mkormendy** [comment, marked as answer]
If your end goal is to control your Konnected alarm system with HomeKit, it's as simple as removing Hubitat's control for Konnected exclusively, and using Homebridge's control instead. It works very well/directly afterwards - considering Homebridge is meant for HomeKit in the first place.

To do that is relatively simple:

1. remove all of Hubitat's configuration and setup on Hubitat itself
2. factory reset the Konnected panel
3. boot up Homebridge and do the minimal configuration for the Homebridge Konnected plugin
4. setup all of your zones by editing the config.json or using the Config UI X plugin graphic user interface to configure the plugin/zone settings

Also see this post if you are having problems with the spinning status of the alarm: https://github.com/mkormendy/homebridge-konnected/issues/25

**Reply by lnjustin** [reply]
Yes, that's one goal, but my primary automation platform is Hubitat. So there's competing interests here it seems. I'll most likely resort to integrating Konnected into Hubitat, and then from Hubitat into HomeKit, and hopefully figure out how to get it to show up in HomeKit correctly.

**Reply by mkormendy** [reply]
Well, I get that it'd be nice to have one platform to control all but that's just not realistic in our IoT world.

Why not have one do some things, and the other do another thing? Then your interests aren't competing, they're working alongside together.

I have a few different platforms Homebridge, Homey, Scrypted for HomeKit Secure Video,... etc.
All which end up interoping with HomeKit as my main interface to control things at the end of the day.

Some bridge-platforms operate faster than others for specific things.
